Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -309,31 +321,49 @@ public CsharpCompiler(ErrorReporter reporter)
       Contract.Requires(dt != null);
       Contract.Requires(wr != null);
 
-      // public abstract class Dt<T> {  // for record types: drop "abstract"
+      // public interface _IDt<T> { // T has variance modifier
+      //   _IDt<T> _Get();  // for co-datatypes
+      //
+      //   bool is_Ctor0 { get; }
+      //   ...
+      //
+      //   T0 dtor_Dtor0 { get; }
+      //   ...
+      //
+      //   // Members that don't violate C# variance restrictions
```

Comment:
If this is the Dafny program:

```
  public interface IOption<out T> {
    T UnwrapOr(T @default);
    ...
  }
```

Where does `UnwrapOr` end up? Can you add the `NeedsCustomReceiver == true` scenario to the example?

Reply:
It can't be in the interface, so you can only access the function through `Option`, where all implementations are located regardless. To make this work, we make it `static` and add `this` as an additional argument. It becomes `T UnwrapOr(IOption<T> _this, T @default)`.